### ISSUE TITLE
azurerm_servicebus_topic forces replacement when namespace_id changes case

### ISSUE THREAD

**Issue by biggles007** [terraform-providers/terraform-provider-azurerm#17762]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.5

### AzureRM Provider Version

3.15.1

### Affected Resource(s)/Data Source(s)

azurerm_servicebus_topic

### Terraform Configuration Files

```hcl
data "azurerm_servicebus_namespace" "test" {
  name                = "xxxxxx"
  resource_group_name = "xxxxx"
}

resource "azurerm_servicebus_topic" "test" {
  name         = "mytopic"
  namespace_id = data.azurerm_servicebus_topic.test.id
}
```

### Debug Output/Panic Output

```shell
n/a
```

### Expected Behaviour

Changing the case on the `name` in the data call should not impact the resource. Topics should not recreate if the case changes in the `namespace_id`.

### Actual Behaviour

The case sensitivity of the `name` in the data call affects the `id` output from the data call. If this changes, it forces a replacement of the topic, even though the `namespace_id` is not case sensitive in the topic resource.

### Steps to Reproduce

Create a Servicebus Namespace for use in the data call
terraform apply
change the case of the `name` in the data call
terraform plan

**Comment by xiaxyi** [comment]
@bigglesuk69 Terraform is case sensitive, different case will be treated as different resource. namespace_id in `azurerm_servicebus_topic` is a forceNew property. This is by-design. Hope it makes sense to you.

**Comment by bigglesuk69** [comment]
My point is, it shouldn't be. Or if you are saying that Terraform **must** be case-sensitive, the azurerm_servicebus_namespace resource **should** return the `resourceid` returned by Azure and not the calculated one. This would resolve some issues and we wouldn't have these issue, as using references from the deployed resource or data calls would always return the right case. Otherwise you end up with resources deployed using the wrong case, if you want to fix the case, it would destroy the existing ones without a large mount of state work.

In other resources there are things that aren't case sensitive (#15549), why are you enforcing a case sensitive operation on a non-case sensitive property to cause a destructive event? This make no-sense what so ever.

**Comment by xiaxyi** [comment]
the reason why terraform is setting the calculated id into state is azure is not case sensitive, which means, if we take whatever value that's returned by azure, the user may face id diff if azure changes the segment, for example, from serviceBus to ServiceBus. we are making things stay the same during the whole life cycle. not sure if I'm understanding your point correctly.

**Comment by bigglesuk69** [comment]
If the data call is incorrect, that should be fixed. So, if I pass `myServiceBUS` into the data call but the resource is `myServiceBus` then if using the `id` from the Azure API call shows a difference, the data call can be amended.

You state that Terraform is case-sensitive and Azure isn't, you **must** validate then generated `id` against the `id` returned from Azure. You can't have it case-sensitive in one part and then non-case sensitive in another.

A resource was recently lost from state for some reason, the engineer imported the resource ID (the one in the Azure portal) and then ran a deployment. As the case sensitivity was out, it destroyed the resource and recreated it.

I'm struggling to see how you can't see the issue with the Terraform generated ID's and actual Azure ID's being different is not a problem. They **should** always match.

**Comment by bigglesuk69** [comment]
@xiaxyi, @tombuildsstuff made a similar suggestion to me re: validating the resource ID being correct on the following issue
https://github.com/hashicorp/terraform-provider-azurerm/issues/5455#issuecomment-576681828

**Comment by xiaxyi** [comment]
@bigglesuk69 yes, exactly, this is how we treat the ID inconsistence in Terraform, otherwise, user may face diff once the case is changed from the azure side. hope you can understand.
